Automattic/jetpack
# ci: Check plugin monorepo dep versions

#### Changes proposed in this Pull Request:

In Jetpack 11.3-beta3 we accidentally had some packages locked to "dev-trunk" instead of the expected release versions. This adds a CI check to try to catch when that happens.

For pushes to trunk and PRs for merge into trunk, all monorepo packages are expected to be locked to dev-trunk in all plugins.

For pushes to a release branch (and any PRs against release branches, should there be any), all monorepo packages for the plugin associated with that branch are expected to be locked to a non-dev version that compares greater than or equal to the latest version in the package's changelog.

#### Other information:


#### Jetpack product discussion


p9dueE-5JX-p2 among other places

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* See what happened in CI for this PR when this ran before #26072 fixed various deps.
  * [x] https://github.com/Automattic/jetpack/runs/8254993590?check_suite_focus=true
  * [x] Reported various plugins incorrectly using "^1.2" style deps. Job failed.
* See what happened in CI for this PR when this ran after being merged with #26072.
  * [x] https://github.com/Automattic/jetpack/runs/8277443101?check_suite_focus=true
  * [x] Job passed.
* Check out f18081a5b, then cherry-pick 7bc6ee12. Run `tools/check-plugin-monorepo-deps.sh --release jetpack`.
  * [x] Reports the problems with the my-jetpack and connection package deps. Exits with a non-zero exit code.
* Create a fake Jetpack release branch that's at f18081a5b with 7bc6ee12 cherry-picked. (Could do it on a forked repo to avoid polluting the monorepo and mirror)
  * [x] On a fork: https://github.com/anomiex/jetpack/actions/runs/3016998911
  * [x] Test runs on the push to the branch, and fails.
* Check out origin/jetpack/branch-11.3, then cherry-pick 7bc6ee12. Run `tools/check-plugin-monorepo-deps.sh --release jetpack`.
  * [x] Reports no problems. Exits with a zero exit code.
